### ISSUE TITLE
[Feature Request] Warn if mutating a proxy itself that is used

### ISSUE THREAD

**Issue by dai-shi** [pmndrs/eslint-plugin-valtio#2]
From https://github.com/pmndrs/valtio/issues/76#issuecomment-772091544

```js
let state = proxy(['hello'])

  // in component
  const snap = useProxy(state)
  const handleClick = () => {
    state = ['hello', 'world']
  }
```

```js
const state = proxy({ arr: ['hello'] })

  // in component
  const snap = useProxy(state.arr)
  const handleClick = () => {
    state.arr = ['hello', 'world']
  }
```

These are patterns that behave unexpectedly.

**Comment by Aslemammad** [comment]
Where exactly the warning should be?

**Comment by dai-shi** [comment]
"mutating a proxy object itself. this might not be expected as it's not reactive."

there can be false positives though. should be good for now.

**Comment by dai-shi** [comment]
Should be fixed in v0.2.0.
